homeassistant/components/palazzetti/climate.py

```diff
@@ -0,0 +1,159 @@
+"""Support for Palazzetti climates."""
+
+from typing import Any, cast
+
+from pypalazzetti.exceptions import CommunicationError, ValidationError
+
+from homeassistant.components.climate import (
+    ClimateEntity,
+    ClimateEntityFeature,
+    HVACMode,
+)
+from homeassistant.const import ATTR_TEMPERATURE, CONF_MAC, CONF_NAME, UnitOfTemperature
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import HomeAssistantError, ServiceValidationError
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from . import PalazzettiConfigEntry
+from .const import (
+    ACTION_NOT_UNAVAILABLE,
+    DOMAIN,
+    FAN_AUTO,
+    FAN_HIGH,
+    FAN_MODES,
+    FAN_SILENT,
+    PALAZZETTI,
+)
+from .coordinator import PalazzettiDataUpdateCoordinator
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: PalazzettiConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up Palazzetti climates based on a config entry."""
+    async_add_entities(
+        [
+            PalazzettiClimateEntity(
+                coordinator=entry.runtime_data,
+            )
+        ]
+    )
+
+
+class PalazzettiClimateEntity(
+    CoordinatorEntity[PalazzettiDataUpdateCoordinator], ClimateEntity
+):
+    """Defines a Palazzetti climate."""
+
+    _attr_has_entity_name = True
+    _attr_hvac_modes = []  # The available modes will be set when we know the current state
```

Comment:
Also, why don't we know the current state yet at creation? The integration already has a state when adding this one right?